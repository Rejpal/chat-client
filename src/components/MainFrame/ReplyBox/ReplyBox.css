
.ReplyBox {
  height: 10%;
  min-height: 100px
}
