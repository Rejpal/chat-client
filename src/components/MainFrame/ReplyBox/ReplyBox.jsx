import React, { Component } from 'react'
import { observer, inject } from 'mobx-react'
import './ReplyBox.css'

const ENTER_KEY = 13

type Props = {
  appStore: any
}

export class ReplyBox extends Component<Props> {
  static defaultProps = {
    appStore: null
  }

  constructor () {
    super()
    this.textInput = React.createRef()
  }

  componentDidMount () {
    this.textInput.current.focus()
  }

  handleChangeMessage = (e: React.SyntheticKeyboardEvent) => {
    if (e.charCode === ENTER_KEY) {
      this.handleSend()
      return
    }

    this.props.appStore.updateCurrentMessage(e.key)
  }

  handleSend = () => {
    this.props.appStore.sendMessage()
    this.textInput.current.focus()
  }

  getMessageText (): string {
    const message = this.props.appStore.getCurrentMessageText()
    return message
  }

  render () {
    return (
      <div className='ReplyBoxWrapper'>
        <div className='ReplyBox'>
          <textarea
            ref={this.textInput}
            onKeyPress={this.handleChangeMessage}
            value={this.getMessageText()}
          />
          <button onClick={this.handleSend}>Send</button>
        </div>
      </div>
    )
  }
}

export default inject('appStore')(observer(ReplyBox))
