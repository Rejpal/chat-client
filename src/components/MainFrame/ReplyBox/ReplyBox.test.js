/* eslint-env mocha, jest */
import React from 'react'
import { ReplyBox } from './ReplyBox'
import { mount } from 'enzyme'

describe('<ReplyBox>', () => {
  it('should render', () => {
    const appStore = {
      getCurrentMessageText: () => { return '' }
    }
    expect(mount(<ReplyBox appStore={appStore} />)).toMatchSnapshot()
  })

  it('should render with text in the input', () => {
    const appStore = {
      getCurrentMessageText: () => { return 'test text' }
    }
    expect(mount(<ReplyBox appStore={appStore} />)).toMatchSnapshot()
  })

  it('should call update upon change', () => {
    const updateMessage = jest.fn()
    const appStore = {
      getCurrentMessageText: () => { return 'test text' },
      updateCurrentMessage: updateMessage
    }
    const wrapper = mount(<ReplyBox appStore={appStore} />)
    wrapper.find('textarea').simulate('keyPress', { key: 's', charCode: 45 })
    expect(updateMessage).toHaveBeenCalled()
  })

  it('should call send on click', () => {
    const handleSendMessage = jest.fn()
    const appStore = {
      getCurrentMessageText: () => { return 'test text' },
      sendMessage: handleSendMessage
    }
    const wrapper = mount(<ReplyBox appStore={appStore} />)
    wrapper.find('button').simulate('click')
    expect(handleSendMessage).toHaveBeenCalled()
  })

  it('should call send on press Enter key', () => {
    const handleSendMessage = jest.fn()
    const appStore = {
      getCurrentMessageText: () => { return 'test text' },
      sendMessage: handleSendMessage
    }
    const wrapper = mount(<ReplyBox appStore={appStore} />)
    wrapper.find('textarea').simulate('keyPress', { key: 'enter', charCode: 13 })
    expect(handleSendMessage).toHaveBeenCalled()
  })
})
