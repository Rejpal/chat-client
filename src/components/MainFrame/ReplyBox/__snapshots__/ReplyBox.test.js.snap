// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReplyBox> should render 1`] = `
<ReplyBox
  appStore={
    Object {
      "getCurrentMessageText": [Function],
    }
  }
>
  <div
    className="ReplyBoxWrapper"
  >
    <div
      className="ReplyBox"
    >
      <textarea
        onKeyPress={[Function]}
        value=""
      />
      <button
        onClick={[Function]}
      >
        Send
      </button>
    </div>
  </div>
</ReplyBox>
`;

exports[`<ReplyBox> should render with text in the input 1`] = `
<ReplyBox
  appStore={
    Object {
      "getCurrentMessageText": [Function],
    }
  }
>
  <div
    className="ReplyBoxWrapper"
  >
    <div
      className="ReplyBox"
    >
      <textarea
        onKeyPress={[Function]}
        value="test text"
      />
      <button
        onClick={[Function]}
      >
        Send
      </button>
    </div>
  </div>
</ReplyBox>
`;
